src/Support/Contracts/GlobalId.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Nuwave\Lighthouse\Support\Contracts;
+
+/**
+ * Encode and decode globally unique IDs.
+ *
+ * The way that IDs are generated basically works like this:
+ *
+ * 1. Take the name of a type, e.g. "User" and an ID, e.g. 123
+ * 2. Glue them together, separated by a colon, e.g. "User:123"
+ * 3. base64_encode the result
+ *
+ * This can then be reversed to uniquely identify an entity in our
+ * schema, just by looking at a single ID.
+ */
+interface GlobalId
+{
+    /**
+     * Glue together a type and an id to create a global id.
+     *
+     * @param  string  $type
+     * @param  string|int  $id
+     * @return string
+     */
+    public function encode(string $type, $id): string;
+
+    /**
+     * Split a global id into the type and the id it contains.
+     *
+     * @param  string  $globalID
+     * @return array Contains [$type, $id], e.g. ['User', '123']
+     */
+    public function decode(string $globalID): array;
```

Comment:
I am not sure that this should really be part of the interface, what do you think @spawnia ?